app/models/miq_provision_virt_workflow.rb

```diff
@@ -577,19 +570,8 @@ def allowed_template_condition
   end
 
   def source_vm_rbac_filter(vms, condition = nil)
-    filter_id = get_value(@values[:vm_filter]).to_i
-    search_options = {:user => @requester, :results_format => :objects}
-    search_options[:conditions] = condition unless condition.blank?
-    template_msg =  "User: <#{@requester.userid}>"
-    template_msg += " Role: <#{@requester.current_group.nil? ? "none" : @requester.current_group.miq_user_role.name}>  Group: <#{@requester.current_group.nil? ? "none" : @requester.current_group.description}>"
-    template_msg += "  VM Filter: <#{@values[:vm_filter].inspect}>"
-    template_msg += "  Passing inital template IDs: <#{vms.collect(&:id).inspect}>" unless vms.blank?
-    _log.info "Checking for allowed templates for #{template_msg}"
-    if vms.empty?
-      MiqSearch.search(filter_id, VmOrTemplate, search_options).first
-    else
-      MiqSearch.filtered(filter_id, VmOrTemplate, vms, search_options)
-    end
+    MiqSearch.filtered(get_value(@values[:vm_filter]).to_i, VmOrTemplate, vms,
+                       :user => @requester, :conditions => condition)
```

Comment:
no longer passing in `[]` to mean search all (which is `targets => nil`), so no need for these checks.

Also, no reason to pull back all these `vms` when `Rbac` is just going to use that information to send a query back to the server to fetch the vms again.